### PR TITLE
lxc: Add support for creating bucket key from yaml (from Incus)

### PR DESCRIPTION
**Pull request by gabrielmougard**
Cherry-picked from https://github.com/lxc/incus/pull/841

### CODE REVIEW

File: test/suites/storage_buckets.sh [hunk 1]
```diff
@@ -164,6 +164,16 @@
...
   rm "${lxdTestFile}"
 
+  # Test creating a bucket key with YAML bucket config.
+  creds=$(lxc storage bucket key create "${poolName}" "${bucketPrefix}.foo" yaml-key < <(cat <<EOF
+role: admin
+access-key: "${bucketPrefix}.foo.ro"
+secret-key: secretpassword
+EOF
+))
+  yamlAccessKey=$(echo "${creds}" | awk '{ if ($1 == "Access" && $2 == "key:") {print $3}}')
```
[the remark this answers is not in the document]
Comment (gabrielmougard) [reply test/suites/storage_buckets.sh:174]: fixed

**Review by tomponline — CHANGES_REQUESTED** [review]
Please can you update tests

**Review by tomponline — DISMISSED** [review]

**Comment by tomponline** [comment]
Just want to get Simon's thoughts on the tests

File: test/suites/storage_buckets.sh [hunk 2]
```diff
@@ -101,6 +101,14 @@ test_storage_buckets() {
   roAccessKey=$(echo "${creds}" | awk '{ if ($1 == "Access" && $2 == "key:") {print $3}}')
   roSecretKey=$(echo "${creds}" | awk '{ if ($1 == "Secret" && $2 == "key:") {print $3}}')
 
+  # Test creating a bucket key with YAML bucket key config.
+  creds=$(lxc storage bucket key create "${poolName}" "${bucketPrefix}.foo" yaml-key < <(cat <<EOF
+role: read-only
+EOF
+))
```
Comment (simondeziel) [inline test/suites/storage_buckets.sh:108]: I think the subshell and cat are not strictly required.
```suggestion
  creds=$(lxc storage bucket key create "${poolName}" "${bucketPrefix}.foo" yaml-key << EOF
role: read-only
EOF
)
```

**Review by simondeziel — APPROVED** [review]
LGTM but I provided a possible simplification in the test part.

File: test/suites/storage_buckets.sh (hunk 2 above)
Comment (gabrielmougard) [reply test/suites/storage_buckets.sh]: fixed

**Review by simondeziel — DISMISSED** [review]

**Review by tomponline — APPROVED** [review]

**Pull request merged**